Plan/common/build.gradle

```diff
@@ -67,38 +67,53 @@ dependencies {
     sqliteDriver "org.slf4j:slf4j-nop:1.7.36"
     ipAddressMatcher "com.github.seancfoley:ipaddress:$ipAddressMatcherVersion"
 
-    shadow "org.apache.commons:commons-text:$commonsTextVersion"
-    shadow "org.apache.commons:commons-compress:$commonsCompressVersion"
-    shadow "commons-codec:commons-codec:$commonsCodecVersion"
-    shadow "com.github.ben-manes.caffeine:caffeine:$caffeineVersion"
-    shadow "com.zaxxer:HikariCP:$hikariVersion"
-    shadow "org.slf4j:slf4j-nop:$slf4jVersion"
-    shadow "org.slf4j:slf4j-api:$slf4jVersion"
-    shadow "com.maxmind.geoip2:geoip2:$geoIpVersion"
-    shadow "com.google.code.gson:gson:$gsonVersion"
-    shadow "org.eclipse.jetty:jetty-server:$jettyVersion"
-    shadow "org.eclipse.jetty:jetty-alpn-java-server:$jettyVersion"
-    shadow "org.eclipse.jetty.http2:http2-server:$jettyVersion"
-    shadow("com.googlecode.json-simple:json-simple:1.1.1") { // json simple used by UUIDFetcher
-        // json-simple has junit (a test dependency) compile scoped
-        exclude group: "junit", module: "junit"
-    }
+    compileOnlyApi "org.jetbrains:annotations:$jetbrainsAnnotationsVersion"
+    api "org.apache.commons:commons-text:$commonsTextVersion"
+    api "org.apache.commons:commons-compress:$commonsCompressVersion"
+    api "commons-codec:commons-codec:$commonsCodecVersion"
+    api "com.github.ben-manes.caffeine:caffeine:$caffeineVersion"
+    implementation "com.zaxxer:HikariCP:$hikariVersion"
+    implementation "org.slf4j:slf4j-nop:$slf4jVersion"
+    implementation "org.slf4j:slf4j-api:$slf4jVersion"
+    implementation "com.maxmind.geoip2:geoip2:$geoIpVersion"
+    api "com.google.code.gson:gson:$gsonVersion"
+    api "org.eclipse.jetty:jetty-server:$jettyVersion"
+    implementation "org.eclipse.jetty:jetty-alpn-java-server:$jettyVersion"
+    implementation "org.eclipse.jetty.http2:http2-server:$jettyVersion"
     implementation "org.jasypt:jasypt:$jasyptVersion:lite"
 
-
     // Swagger annotations
     implementation "jakarta.ws.rs:jakarta.ws.rs-api:4.0.0"
+    api "io.swagger.core.v3:swagger-annotations:$swaggerVersion"
     implementation "io.swagger.core.v3:swagger-core-jakarta:$swaggerVersion"
     implementation "io.swagger.core.v3:swagger-jaxrs2-jakarta:$swaggerVersion"
 
+
+    // Test Tooling Dependencies
+    testFixturesApi "org.junit.jupiter:junit-jupiter:$junitVersion"      // JUnit 5
+    testRuntimeOnly "org.junit.platform:junit-platform-launcher"
+
+    testFixturesApi "org.mockito:mockito-core:$mockitoVersion"           // Mockito Core
+    testFixturesApi "org.mockito:mockito-junit-jupiter:$mockitoVersion"  // Mockito JUnit 5 Extension
+    testFixturesImplementation "net.playeranalytics:platform-abstraction-layer-api:$palVersion"
```

Comment:
I'm still trying to wrap my head around textFixtures so a couple of questions.

- Why is pal-api needed in textFixtures when it is also declared as `implementation`? Is it so that other modules using the `textFixtures(:common)` don't need to declare pal-api as `testImplementation` separately 
- don't they already have pal-api in test class path from the transitive dependencies of `implementation pal-bukkit` for example?

Reply:
pal-api is needed as a testFixturesImplementation because testFixtures are a bit more separated from the normal dependency scopes ("improved separation of concerns" -Gradle docs). Implementation dependencies aren't included on its compile classpath (unlike in the normal testCompileClasspath, for example). 

So yes, in the platform modules, PAL is an implementation dependency and their testCompileClasspath gets the dependency from that.